### PR TITLE
AppendBlobClient.getOutputStream(boolean) to overwrite correctly if set to true

### LINKED ISSUE #32602: [QUERY] `AppendBlobClient.getBlobOutputStream` overwrites or not?
**Query/Question**
The Javadoc of `AppendBlobClient.getBlobOutputStream` states that in case the BLOB already exists it will be overwritten:
https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.20.1/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java#L102-L103

But the implementation looks like the BLOB will be appended to:
https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.20.1/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java#L165-L167

Does it overwrite or append?

### PR DESCRIPTION
**Pull request by ibrahimrabab**
# Description

resolves #32602

When `AppendBlobClient.getOutputStream(true)` is called, the existing blob is not overwritten. Instead, new data gets appended to the existing data. 

Changed javadoc for `AppendBlobClient.getOutputStream()` that specifies new data will be appended to the existing blob (which is the actual correct behavior) instead of getting overwritten.

Changed logic for `AppendBlobClient.getOutputStream(boolean)` to create a new blob that will overwrite existing blob if `overwrite=true`. 
# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java [hunk 1]
```diff
@@ -108,6 +111,28 @@ public BlobOutputStream getBlobOutputStream() {
...
+     * <p>
+     *
+     * @return A {@link BlobOutputStream} object used to write data to the blob.
+     * @param overwrite Whether an existing blob should be deleted and recreated, should data exist on the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public BlobOutputStream getBlobOutputStream(boolean overwrite) {
+        AppendBlobRequestConditions requestConditions = null;
+        if (!overwrite) {
+            if (exists()) {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java:125]: Do we want to throw here or put the OutputStream into append mode?

Comment (ibrahimrabab) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java:125]: So it looks like in [.NET](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs#L1984) when overwrite is false and blob exists, they continue to append. We should probably follow the same logic.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java [hunk 2]
```diff
@@ -109,8 +111,25 @@ public BlobOutputStream getBlobOutputStream() {
     }
 
     /**
-     * Creates and opens an output stream to write data to the append blob. If the blob already exists on the service,
-     * it will be overwritten.
+     * Creates and opens an output stream to write data to the append blob.
+     *
+     * @return A {@link BlobOutputStream} object used to write data to the blob.
+     * @param overwrite Whether an existing blob should be deleted and recreated, should data exist on the blob.
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java:117]: I'd mention in the description section how true and false correlate to behavior.

**Review by alzimmermsft — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**